File: app/src/main/java/de/hype/hypenotify/services/WifiConnectionReceiver.java
package de.hype.hypenotify.services;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.net.wifi.WifiManager;

public class WifiConnectionReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (WifiManager.NETWORK_STATE_CHANGED_ACTION.equals(intent.getAction())) {
            onWifiConnected(context);
        }
    }

    private void onWifiConnected(Context context) {
        // Your code to handle Wi-Fi connection
    }
}

File: app/src/main/java/de/hype/hypenotify/app/services/WifiConnectionReceiver.kt
package de.hype.hypenotify.app.services

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.wifi.WifiManager

class WifiConnectionReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent) {
        if (WifiManager.NETWORK_STATE_CHANGED_ACTION == intent.getAction()) {
            onWifiConnected(context)
        }
    }

    private fun onWifiConnected(context: Context?) {
        // Your code to handle Wi-Fi connection
    }
}